hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/UploadArtifactView.java

```diff
@@ -260,4 +270,30 @@ void destroy() {
             artifactDetailsGridLayout.unsubscribeListener();
         }
     }
+
+    @Override
+    public void beforeLeave(final ViewBeforeLeaveEvent event) {
+        if (isAnyUploadInUploadQueue()) {
+            final ConfirmationDialog confirmDeleteDialog = new ConfirmationDialog(i18n,
+                    i18n.getMessage(UIMessageIdProvider.CAPTION_FILE_UPLOAD_INPROGRESS),
+                    i18n.getMessage(UIMessageIdProvider.MESSAGE_CLEAR_FILE_UPLOAD_QUEUE), ok -> {
+                        if (Boolean.TRUE.equals(ok)) {
+                            // Clear all queued file uploads
+                            artifactUploadState.clearFileStates();
+                            event.navigate();
+                        }
+                    }, UIComponentIdProvider.UPLOAD_QUEUE_CLEAR_CONFIRMATION_DIALOG);
+            UI.getCurrent().addWindow(confirmDeleteDialog.getWindow());
+            confirmDeleteDialog.getWindow().bringToFront();
+        } else {
+            event.navigate();
+        }
+    }
+
+    private boolean isAnyUploadInUploadQueue() {
+        return artifactUploadState.getAllFileUploadProgressValuesFromOverallUploadProcessList().stream()
+                .map(FileUploadProgress::getFileUploadStatus)
+                .anyMatch(fileUploadStatus -> fileUploadStatus.equals(UPLOAD_STARTED));
```

Comment:
Shouldn't we also check for the status `UPLOAD_IN_PROGRESS`?

Reply:
The files with an upload status of `UPLOAD_IN_PROGRESS` will still be uploaded, even when the view is changed. Only queued up (and not yet started) file uploads will be removed.